docs/design/wildcard-namespace-design.md

```diff
@@ -0,0 +1,97 @@
+# Design proposal: Wildcard Namespace Support for Velero Backup CLI
+
+## Abstract
+
+This proposal enables full wildcard namespace support for Velero backup CLI commands to resolve patterns like `prod-*` or `*-staging` into actual namespace names during command execution.
+Currently, wildcard patterns are stored literally in backup specs causing backup failures when the system tries to list resources in non-existent namespaces.
+
+## Background
+
+Velero's backup CLI currently accepts wildcard patterns for namespace inclusion and exclusion but has incomplete implementation.
+When users run commands like `velero backup create test --include-namespaces "prod-*"`, the wildcard pattern is stored literally in the backup specification without being resolved to actual namespace names.
+
+This causes backup failures because the system attempts to list resources in namespaces literally named `"prod-*"` which don't exist.
+The disconnect occurs between namespace collection (which uses proper glob matching) and resource collection (which uses literal patterns), resulting in inconsistent behavior and silent failures.
+
+## Goals
+
+- Enable wildcard patterns in backup CLI commands to work correctly by resolving them to actual namespace names
```

Comment:
We probably need the wildcard expansion to happen in the Backup schedule as well. Basically whatever logic we put in the CLI to generate the backup.spec.Included/ExcludedNamespaces field values should also be added to the schedule controller where that controller creates a Velero backup CR from the Schedule template. This still preserves the existing backup functionality, since backup processing itself won't be doing the expansion.